src/WasmParser.ts

```diff
@@ -2073,7 +2073,12 @@ export class BinaryReader {
   }
 
   private readCodeOperator_0xfc(): boolean {
-    var code = this._data[this._pos++] | 0xfc00;
+    var pos = this._pos;
+    // u32 must not exceed ceil(N/7) = 5 bytes, + 1 prefix byte.
+    if (!this._eof && pos + 1 + 5 > this._length) {
```

Comment:
`this.hasVarIntBytes()` needs to be used here instead (no `this._eof`-padding check is needed)

Reply:
also _xfd and _xfe missing this check too.

Reply:
0xfd and 0xfe have a check for MAX_CODE_OPERATOR_?_SIZE, do they still need to be around if we also have with `hasVarIntBytes`? (With leb128 opcodes, those MAX_CODE_OPERATOR_?_SIZE needs to be updated too.)

Reply:
>  those MAX_CODE_OPERATOR?_SIZE needs to be updated too.

Not really, these are "does it need more bytes check" for chunked parsing. So it rather checks if more data is need and unrelated to the subject.

> `hasVarIntBytes`

Agree that we need separate less generic hasVar32Bytes for check.

Reply:
Okay, I pushed commit to fix what I mentioned above.